tests/test_signal_generator.cpp

```diff
@@ -0,0 +1,169 @@
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/DigitalConverterComponent.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+#include <hyro/test/ComponentTester.h>
+
+#include <gtest/gtest.h>
+
+#define TEST_AMPLITUDE 100.0
+#define TEST_FREQUENCY 30.0
+#define TEST_COSINE false
+#define TEST_TIMESTAMP 1000000000
+#define TEST_FRAMID     "Teste 1"
+#define TEST_VALUE      20.0
+#define TEST_RETURNED_VALUE_POS 1.0
+#define TEST_RETURNED_VALUE_NEG 0.0
+
+static void
+__assert_and_run_state_machine(hyro::StateMachine &sm,
+                               const hyro::ComponentConfiguration &conf,
+                               const hyro::ConnectionConfiguration &endpoint)
+{
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.init(conf));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.start());
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.connect(endpoint));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.check());
+}
+
+using namespace signal_generator;
+using namespace digital_converter;
+
+namespace hyro
+{
+namespace test
+{
+
+TEST(ExampleTest, SignalGeneratorComponentCheck)
```

Comment:
@rfcarneiro this test case is good. However, I would verify not only if the value is between **-amplitude** and **+amplitude**. I would check try to something like is shown below:

```c++
ASSERT_NEAR(value_analog_out->value, expected_value, 0.001);
```

Reply:
For a real test I should get a full period and then check the waveform.
As this is only a practice. I will let it the way it is.